compiler_opt/es/inlining/inlining_worker.py

```diff
@@ -0,0 +1,124 @@
+  """
+
+  def _setup_base_policy(self):
+    self._tf_base_temp_dir = tempfile.mkdtemp()
+    policy = policy_utils.create_actor_policy()
+    saver = policy_saver.PolicySaver({"policy": policy})
+    saver.save(self._tf_base_temp_dir)
+    self._tf_base_policy_path = os.path.join(self._tf_base_temp_dir, "policy")
+
+  def __init__(
+      self,
+      *,
+      gin_config: str,
+      clang_path: str,
+      llvm_size_path: str,
+      thread_count: int = 128,
+  ):
+    """Initializes the InliningWorker class."""
+    gin.parse_config(gin_config)
+    self._setup_base_policy()
+    self._inliner = inlining_runner.InliningRunner(
+        clang_path=clang_path,
+        llvm_size_path=llvm_size_path,
+    )
+    self._thread_count = thread_count
+
+  # Deletion here is best effort as it occurs at GC time. If the shutdown is
+  # forced, cleanup might not happen as expected. This does not matter too
+  # much though as resource leakage will be small, and any cloud setups will
+  # have tempdirs wiped periodically.
+  def __del__(self):
+    shutil.rmtree(self._tf_base_temp_dir, ignore_errors=True)
+
+  def _compile_module_and_get_size(
+      self,
+      loaded_module_spec: corpus.LoadedModuleSpec,
+      output_directory: str,
+      tflite_policy_path: str | None,
+  ) -> float:
+    """Compiles a single LoadedModuleSpec and returns its native code size."""
+    working_dir = tempfile.mkdtemp(dir=output_directory)
+
+    # Build the final command line using LoadedModuleSpec
+    original_cmd_line = loaded_module_spec.build_command_line(working_dir)
+
+    result = self._inliner.compile_fn(original_cmd_line, tflite_policy_path,
+                                      True, working_dir)
+
+    return result[inlining_runner.DEFAULT_IDENTIFIER][1]
+
+  def compile(self, policy: bytes | None,
+              modules: list[corpus.LoadedModuleSpec]) -> float | None:
+    with tempfile.TemporaryDirectory() as compilation_dir:
+      tflite_policy_path = None
+      if policy is not None:
+        tflite_policy_path = policy_utils.convert_to_tflite(
+            policy, compilation_dir, self._tf_base_policy_path)
+
+      with concurrent.futures.ThreadPoolExecutor(
+          max_workers=self._thread_count) as thread_pool:
+        compile_futures = {
+            thread_pool.submit(
+                self._compile_module_and_get_size,
+                module,
+                compilation_dir,
+                tflite_policy_path,
+            ):
+                module for module in modules
+        }
+
+        total_size = 0
+        try:
+          for future in concurrent.futures.as_completed(compile_futures):
+            size = future.result()
+            # Check for failure indicator from the compile function
+            if size == float("inf"):
+              raise ValueError(
+                  "Size obtained is infinity. This is not expected.")
+            total_size += size
+        except Exception as e:
```

Comment:
This exception is way too broad. It should probably be scoped to something like `CalledProcessError`.